### PR TITLE
enhance: optimize some cache to reduce memory usage

### PR DESCRIPTION
**Pull request by zhagnlu**
#33533

### CODE REVIEW

**Comment by zhagnlu** [comment]
rerun ut

File: internal/core/src/segcore/InsertRecord.h [hunk 1]
```diff
@@ -285,13 +286,13 @@ class OffsetOrderedArray : public OffsetMap {
 
  private:
     bool is_sealed = false;
-    std::vector<std::pair<T, int64_t>> array_;
+    std::vector<std::pair<T, int32_t>> array_;
```
Comment (longjiquan) [inline internal/core/src/segcore/InsertRecord.h:289]: Good catch.

**Review by longjiquan — COMMENTED** [review]
/lgtm

**Pull request merged**